### DISCUSSION TITLE
Constant external download traffic, v2.9.5

### DISCUSSION THREAD

**Discussion by epc002** [codeproject/CodeProject.AI-Server#234] [Q&A] [answered]
Recently +updated from 2.6.5 to 2.9.5, and am noticing that when the CPAI dashboard is open there is constant external WAN network traffic, approx 250kbps download and 50kbps upload.  Closing the dashboard and it goes away.  Before I resort to a wireshark session, anyone know what this traffic is for and why it is necessary?

**Comment by Dmeyer07** [comment]
Hmm interesting. I'll have to take a look when I get home and see if this is happening on mine as well. Did you get a chance to fire up wireshark and take a look at what it's trying to accomplish?

**Reply by epc002** [reply]
I took a quick stab at Wireshark with GeoIP, looks like whenever the CPAI2.9.5 dashboard is open, there's constant traffic with AS54113, 185.199.109.153, which is netname US-GITHUB-20170413, located in San Francisco. It's primarily download traffic, about 160x download vs upload.

**Reply by VorlonCD** [reply]
Can confirm with 2.9.4, downloading much more than uploading from 185.199.109.153 / `cdn-185-199-109-153.github.com`.

Fiddler is only showing traffic for LOCAL firefox traffic, but in System Informer I see the download traffic from `codeproject.ai.server.exe`.

If I ping `codeproject.github.io` its **the SAME IP address**.

Only while the web interface is open.   Connection drops when you close the tab.

At first I thought it was just downloading a module in the background but they have all been downloaded.

I think something in a loop is mistakenly always re-downloading some json config file?

**Reply by VorlonCD** [reply]
Ok, definitely a bug.   Someone should log over in Issues.    I couldnt get wireshark to decrypt TLS for some reason, but finally got Fiddler to do it by stopping the service and running the EXE directly as administrator.

It appears to be re-downloading modules.json (5 kb) exactly every 5 seconds, so that should help narrow down in the code.

![image](https://github.com/user-attachments/assets/ed9652b4-16f1-4bcf-a4bd-b73279e0ff61)

Maybe coming from `ModuleInstaller.cs > GetInstallableModulesAsync` or GetDownloadableModelsAsync?    But it appears to have a timespan check much higher than 5 seconds?

**Reply by epc002** [reply]
logged to "issues"

**Comment by ChrisMaunder** [comment]
Here's the setup for the polling of services:

https://github.com/codeproject/CodeProject.AI-Server/blob/e57de24e22325c400e4c4ad6dcdd93b971415542/src/server/wwwroot/index.html#L40

The dashboard constantly hits the server for logs, status etc (a TODO was to switch to Signal-R) and we also hit our github.io pages to check for new versions of the server, and new versions of modules (though this should be [once an hour](https://github.com/codeproject/CodeProject.AI-Server/blob/e57de24e22325c400e4c4ad6dcdd93b971415542/src/server/wwwroot/assets/dashboard.js#L2C67-L3C1)

We also send UDP packets for the mesh processing. Try going to the mesh tab and turning off mesh processing.

**Reply by VorlonCD** [reply]
Mesh was off when I tested.    From watching Fiddler, Modules.json is definitely being re-downloaded from github.io exactly every 5 seconds when the web UI is open.

**Reply by Dmeyer07** [reply]
Seems like a lot of unnecessary traffic for something that's locally hosted.. 
Hourly checks for version updates even seems excessive?

**Comment by ChrisMaunder** [comment, marked as answer]
Found the issue. The freq for checking downloadable modules was set as the same as the freq for checking live module status. Downloadable modules requires a check of the modules.json file on github,io, whereas module statuses is a local call to the server.

I've fixed the code and pushed. v2.9.8 will have this correction

**Reply by epc002** [reply]
Thanks Chris!
